app/controllers/ranks_controller.rb

```diff
@@ -0,0 +1,21 @@
+
+class RanksController < ApplicationController
+  def update_ranks
+    params.require [:users, :realm, :platform]
+    users = params[:users]
+    realm = params[:realm]
+    platform = params[:platform]
+
+    rank_rows = []
+    Identity.where(identifier: users.keys, platform: platform).find_each do |identity|
+      group = users[identity.identifier]['group']
+      next unless group
+
+      rank_rows << {name: group, user_id: identity.user_id, realm: realm}
+    end
+
+    Rank.import rank_rows, on_duplicate_key_update: [:name] unless rank_rows.length == 0
+
+    render plain: 'Ranks updated', status: 202
```

Comment:
Let's return a `201 Created` here instead.

Initially I thought that we'd break early, return a `202 Accepted` and then have a thread handle the batch processing (I like this idea better, but this is fine for now).

`201` means "I got it, processed it, and created the things you wanted"
`202` means "I got it, and it will be processed at some point in the future"